src/datasets/features/audio.py

```diff
@@ -0,0 +1,38 @@
+from dataclasses import dataclass, field
+from typing import Any, ClassVar, Optional
+
+import pyarrow as pa
+
+
+@dataclass
+class Audio:
+    """Audio Feature to extract audio data from an audio file.
+
+    Args:
+        sampling_rate (:obj:`int`, optional): Target sampling rate. If `None`, the native sampling rate is used.
+    """
+
+    sampling_rate: int = None
+    id: Optional[str] = None
+    # Automatically constructed
+    dtype: ClassVar[str] = "dict"
+    pa_type: ClassVar[Any] = None
+    _type: str = field(default="Audio", init=False, repr=False)
+
+    def __call__(self):
+        return pa.string()
+
+    def decode_example(self, value):
+        """Decode example audio file into audio data.
+
+        Args:
+            value: Audio file path.
+
+        Returns:
+            dict
+        """
+        import librosa
+
+        with open(value, "rb") as f:
+            array, sample_rate = librosa.load(f, sr=self.sampling_rate)
```

Comment:
Really like the ability to resample audio upon reading the file, this covers one of the main pain points :) 
You can also add `librosa.load(..., mono=self.mono)` to convert stereo clips and support pretty much all kinds of datasets.

Reply:
@anton-l, the parameter `mono` in `librosa` performs the mean along the channel axis. Is it enough to have that or would it be useful to have more flexibility on this?

Reply:
Averaging the channels is the most widespread way of mixing to mono, yes. The problems with that approach are very rare (mostly with some music tracks doing "out of phase stereo").
Anyway, if the users want to mix the audio manually, they can just set `mono=False` and handle the channels later in the pipeline.